src/mongo_client.ts

```diff
@@ -295,7 +298,7 @@ export class MongoClient extends EventEmitter implements OperationParent {
   }
 
   get readPreference(): ReadPreference {
-    return ReadPreference.primary;
+    return this.s.readPreference;
```

Comment:
I know this is a port, so sorry about not bringing this up on the original, but it occurred to me that this didn't effect any tests? should we throw in a quick 'respects user provided preference' test?

Reply:
did write 3.6 test, missed it in port TY!